src/repr/benches/packed.rs

```diff
@@ -37,28 +75,174 @@ fn bench_interval(c: &mut Criterion) {
             std::hint::black_box(normal);
         })
     });
 
     group.finish()
 }
 
-criterion_group!(benches, bench_interval, bench_time);
+fn bench_acl_item(c: &mut Criterion) {
+    let mut group = c.benchmark_group("AclItem");
+    group.throughput(Throughput::Elements(1));
+
+    let acl_item = AclItem {
+        grantee: Oid(1),
+        grantor: Oid(2),
+        acl_mode: AclMode::all(),
+    };
+    group.bench_function("encode", |b| {
+        let mut buf = vec![0u8; 32];
+        b.iter(|| {
+            let packed = PackedAclItem::from_value(std::hint::black_box(acl_item));
+            std::hint::black_box(&mut buf[..PackedAclItem::SIZE])
+                .copy_from_slice(packed.as_bytes());
+        })
+    });
+    group.bench_function("encode/proto", |b| {
+        let mut total_duration = Duration::from_secs(0);
+        let mut buf = vec![0u8; 64];
+
+        b.iter_custom(|iters| {
+            for _ in 0..iters {
+                let start = Instant::now();
+                let proto = std::hint::black_box(acl_item).into_proto();
+                proto.encode(std::hint::black_box(&mut buf)).unwrap();
+                let elapsed = start.elapsed();
+
+                buf.clear();
+                total_duration += elapsed;
+            }
+            total_duration
+        })
+    });
+
+    const PACKED: [u8; 16] = [0, 0, 0, 1, 0, 0, 0, 2, 0, 0, 0, 0, 224, 0, 3, 15];
+    group.bench_function("decode", |b| {
+        b.iter(|| {
+            let packed = PackedAclItem::from_bytes(std::hint::black_box(&PACKED)).unwrap();
+            let normal = std::hint::black_box(packed).into_value();
+            std::hint::black_box(normal);
+        })
+    });
+    const ENCODED: [u8; 12] = [8, 1, 16, 2, 26, 6, 8, 143, 134, 128, 128, 14];
+    group.bench_function("decode/proto", |b| {
+        b.iter(|| {
+            let proto = ProtoAclItem::decode(std::hint::black_box(&ENCODED[..])).unwrap();
+            let normal: AclItem = std::hint::black_box(proto).into_rust().unwrap();
+            std::hint::black_box(normal);
+        })
+    });
+
+    group.finish();
+}
+
+fn bench_mz_acl_item(c: &mut Criterion) {
+    let mut group = c.benchmark_group("MzAclItem");
+    group.throughput(Throughput::Elements(1));
+
+    let acl_item = MzAclItem {
+        grantee: RoleId::User(1),
+        grantor: RoleId::User(2),
+        acl_mode: AclMode::all(),
+    };
+    group.bench_function("encode", |b| {
+        let mut buf = vec![0u8; 32];
+        b.iter(|| {
+            let packed = PackedMzAclItem::from_value(std::hint::black_box(acl_item));
+            std::hint::black_box(&mut buf[..PackedMzAclItem::SIZE])
+                .copy_from_slice(packed.as_bytes());
+        })
+    });
+    group.bench_function("encode/proto", |b| {
+        let mut total_duration = Duration::from_secs(0);
+        let mut buf = vec![0u8; 64];
+
+        b.iter_custom(|iters| {
```

Comment:
Doesn't total_duration need to reset between `iter_custom` calls? That's what the criterion example seems to do https://docs.rs/criterion/latest/criterion/struct.Bencher.html#method.iter_custom

Reply:
Good call, although when I reset the duration for each call it `iter_custom(...)` it didn't make a difference. In general I probably wasn't using it correctly because with the refactor to use `.iter(...)` the protobuf encoding performance improved